File: Assignment/Abstraction/Expressions/DeclarationNode.cs
namespace Assignment.Abstraction.Expressions
{
    internal struct DeclarationNode: INode
    {
        public DeclarationNode(string name, DataTypes dataType)
        {
            this.Name = name;
            this.DataType = dataType;
        }

        public string Name { get; }
        public DataTypes DataType { get; }

        public string DisplayName()
        {
            return $"{nameof(DeclarationNode)}(name: {this.Name})";
        }
    }
}

File: src/main/java/compiler/abstraction/expressions/DeclarationNode.java
package compiler.abstraction.expressions;

import java.util.UUID;

import compiler.abstraction.DataType;
import compiler.abstraction.INode;

public class DeclarationNode extends INode {
    public final String name;
    public final DataType dataType;

    public DeclarationNode(String name, DataType dataType) {
        super();
        this.name = name;
        this.dataType = dataType;
    }

    public DeclarationNode(String name, DataType dataType, UUID id) {
        super(id);
        this.name = name;
        this.dataType = dataType;
    }

    @Override
    public String getDisplayName() {
        return String.format("%s(name: %s, type: %s)", DeclarationNode.class.getSimpleName(), this.name, this.dataType);
    }
}